packages/react-components/react-field/src/components/Field/useField.tsx

```diff
@@ -2,106 +2,70 @@ import * as React from 'react';
-    validationMessageIcon,
-    validationState,
-    ...restOfProps
-  } = props;
-
-  const fieldProps = {
-    className,
-    control,
-    hint,
-    label,
-    orientation,
-    root,
-    style,
-    validationMessage,
-    validationMessageIcon,
-    validationState,
-  };
-
-  return [fieldProps, restOfProps] as const;
-};
-
 /**
  * Create the state required to render Field.
  *
  * The returned state can be modified with hooks such as useFieldStyles_unstable,
  * before being passed to renderField_unstable.
  *
  * @param props - Props passed to this field
- * @param ref - Ref to the control slot (primary slot)
- * @param params - Configuration parameters for this Field
+ * @param ref - Ref to the root
  */
-export const useField_unstable = <T extends FieldControl>(
-  props: FieldPropsWithOptionalComponentProps<T>,
-  ref: React.Ref<HTMLElement>,
-  params: FieldConfig<T>,
-): FieldState<T> => {
-  const [fieldProps, controlProps] = getPartitionedFieldProps(props);
-  const { orientation = 'vertical', validationState } = fieldProps;
-  const { labelConnection = 'htmlFor', ariaInvalidOnError = true } = params;
+export const useField_unstable = (props: FieldProps, ref: React.Ref<HTMLDivElement>): FieldState => {
+  const { children, orientation = 'vertical', required, validationState, size } = props;
 
   const baseId = useId('field-');
 
-  const root = resolveShorthand(fieldProps.root, {
-    required: true,
-    defaultProps: getNativeElementProps('div', fieldProps),
-  });
+  const root = getNativeElementProps('div', { ...props, ref }, ['children']);
 
-  const label = resolveShorthand(fieldProps.label, {
+  const label = resolveShorthand(props.label, {
     defaultProps: {
       id: baseId + '__label',
-      required: controlProps.required,
-      size: typeof controlProps.size === 'string' ? controlProps.size : undefined,
+      required,
+      size,
       // htmlFor is handled below
     },
   });
 
-  const validationMessage = resolveShorthand(fieldProps.validationMessage, {
+  const validationMessage = resolveShorthand(props.validationMessage, {
     defaultProps: {
       id: baseId + '__validationMessage',
       role: validationState === 'error' ? 'alert' : undefined,
     },
   });
 
-  const hint = resolveShorthand(fieldProps.hint, {
+  const hint = resolveShorthand(props.hint, {
     defaultProps: {
       id: baseId + '__hint',
     },
   });
 
-  const validationMessageIcon = resolveShorthand(fieldProps.validationMessageIcon, {
+  const validationMessageIcon = resolveShorthand(props.validationMessageIcon, {
     required: !!validationState,
     defaultProps: {
       children: validationState ? validationMessageIcons[validationState] : undefined,
     },
   });
 
-  // Hook up aria props on the control
-  if (label && labelConnection === 'aria-labelledby') {
+  const childElement = React.isValidElement(children) ? React.Children.only(children) : undefined;
+  const controlProps = { ...childElement?.props };
```

Comment:
```suggestion
  const controlProps: FieldChildProps = { ...childElement?.props };
```

nit: Explicit typing?

Reply:
Technically its type is inferred because `children` is `React.ReactElement<FieldChildProps>`. That said, that's a long chain of inference, and it doesn't hurt to add an explicit type here. 👍